sources/packages/web/src/views/aest/FormioEditor.vue

```diff
@@ -0,0 +1,305 @@
+    });
+
+    /**
+     * Load the select form definition.
+     * @param form form to be loaded.
+     */
+    const formSelected = async (form: FormAPIOutDTO): Promise<void> => {
+      if (!form?.name) {
+        return;
+      }
+      formDefinitionLoading.value = true;
+      try {
+        const definition = await ApiClient.DynamicForms.getFormDefinition(
+          form.name,
+        );
+        // Recreate the formio builder.
+        builder = await Formio.builder(formioBuilderRef.value, definition, {
+          noNewEdit: true,
+          noDefaultSubmitButton: true,
+          alwaysConfirmComponentRemoval: true,
+          builder: {
+            custom: FORMIO_CUSTOM_COMPONENTS,
+          },
+        });
+        updateFormDefinitionInspector();
+        builder?.on("change", () => {
+          updateFormDefinitionInspector();
+        });
+      } catch {
+        snackBar.error("Unexpected error loading form definition.");
+      } finally {
+        formDefinitionLoading.value = false;
+      }
+    };
+
+    /**
+     * Update the form definition inspector with the current form definition.
+     */
+    const updateFormDefinitionInspector = async (): Promise<void> => {
+      formDefinition.value = getFormattedFormDefinition(builder.schema);
+    };
+
+    /**
+     * Persists the change to the dynamic form definition database.
+     */
+    const saveDynamicForm = async (): Promise<void> => {
+      if (selectedForm.value?.name === undefined) {
+        snackBar.warn("Please select a form to save.");
+        return;
+      }
+      const shouldContinue = await saveDynamicFormModal.value.showModal();
+      if (!shouldContinue) {
+        return;
+      }
+      try {
+        saveDynamicFormLoading.value = true;
+        await ApiClient.DynamicForms.updateForm(selectedForm.value.name, {
+          formDefinition: getReadyToSaveFormDefinition(builder.schema),
+        });
+      } catch {
+        snackBar.error("Unexpected error saving form definition.");
+      } finally {
+        saveDynamicFormLoading.value = false;
+      }
+    };
+
+    /**
+     * Copy the form definition to the clipboard.
+     */
+    const copyToClipboard = async (): Promise<void> => {
+      try {
+        const formDefinition = getFormattedFormDefinition(builder.schema);
+        await navigator.clipboard.writeText(formDefinition);
+        snackBar.success("Form definition copied to clipboard.");
+      } catch {
+        snackBar.error("Error copying form definition to clipboard.");
+      }
+    };
+
+    /**
+     * Open the form definition in the GitHub repository.
+     */
+    const viewOnRepo = async (): Promise<void> => {
+      if (selectedForm.value === undefined) {
+        snackBar.warn("Please select a form.");
+        return;
+      }
+      window.open(
+        `https://github.dev/bcgov/SIMS/blob/main/sources/packages/forms/src/form-definitions/${selectedForm.value.name.toLowerCase()}.json`,
+        "_blank",
+      );
```

Comment:
Sonar use to complain about using `noopener` looks like it stopped now.

![image](https://github.com/user-attachments/assets/5718db49-431a-41c4-b97a-fc8164d27491)

Reply:
I will add.